components/camel-spring-boot/src/main/java/org/apache/camel/spring/boot/SpringTypeConverter.java

```diff
@@ -40,6 +41,11 @@ public SpringTypeConverter(List<ConversionService> conversionServices) {
         if (type.getCanonicalName().startsWith("org.apache")) {
             return null;
         }
+        
+        // do not attempt to convert List -> Map. Ognl expression may use this converter  as a fallback expecting null
+        if (type.isAssignableFrom(Map.class) && (value.getClass().isArray() || value instanceof List)) {
```

Comment:
I wonder if we should Collection instead of List to capture any kind of list like structures.